app/src/main/java/com/example/swugger/AddReminderDialogFragment.java

```diff
@@ -0,0 +1,118 @@
+
+            case R.id.spinner_minutes_new_reminder_dialog:
+                minutesBefore = Integer.parseInt((String) parent.getItemAtPosition(position));
+        }
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView<?> parent) {
+
+    }
+
+    /** MUST CALL
+     *  Attaches this DialogFragment to another DialogFragment (in this case, the edit events DialogFragment. */
+    public void setTargetDialogFragment(EditEventDialogFragment dialogFragment) {
+        mCallback = dialogFragment;
+    }
+
+    /** Implements the AddReminderDialogListener so that any Activity
+     *  that implements it can retrieve information from this Dialog. */
+    public interface AddReminderDialogListener {
+        void onPositiveClickAddReminder(Integer daysBefore, Integer hoursBefore, Integer minutesBefore);
+        void onNegativeClickAddReminder();
+    }
+
+    @Override
+    public Dialog onCreateDialog(final Bundle savedInstanceState) {
+        // Use the Builder class for convenient dialog construction
+        AlertDialog.Builder builder = new AlertDialog.Builder(mCallback.getActivity());
+        // Get the layout inflater
+        LayoutInflater inflater = mCallback.getActivity().getLayoutInflater();
+        // Create the dialog view
+        View root = inflater.inflate(R.layout.dialog_new_reminder, null);
+
+        builder.setView(root)
+                .setPositiveButton(R.string.create_reminder_dialog_accept, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        //TODO: check if there is a valid reminder to create
+                        mCallback.onPositiveClickAddReminder(daysBefore, hoursBefore, minutesBefore);
+                    }
+                })
+                .setNegativeButton(R.string.create_reminder_dialog_cancel, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        // User cancelled the dialog
+                        mCallback.onNegativeClickAddReminder();
+                    }
+                });
+
+        // set spinners for reminders
+        mDaySpinner = (Spinner) root.findViewById(R.id.spinner_days_new_reminder_dialog);
+        daySpinnerAdapter = ArrayAdapter.createFromResource(mCallback.getContext(), R.array.reminder_numbers, android.R.layout.simple_spinner_item);
+        daySpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        mDaySpinner.setAdapter(daySpinnerAdapter);
+        mDaySpinner.setOnItemSelectedListener(this);
+        mDaySpinner.setSelection(DAY_SPINNER_DEFAULT_POS);
+        daysBefore = DAY_SPINNER_DEFAULT_POS;
+
+        mHourSpinner = (Spinner) root.findViewById(R.id.spinner_hours_new_reminder_dialog);
+        hourSpinnerAdapter = ArrayAdapter.createFromResource(mCallback.getContext(), R.array.reminder_numbers, android.R.layout.simple_spinner_item);
```

Comment:
Pretty sure you can create a function that does this in general for Day/Hour/Minute